x-pack/plugins/security_solution/public/common/components/links/index.tsx

```diff
@@ -57,34 +57,36 @@ const UserDetailsLinkComponent: React.FC<{
   isButton?: boolean;
   onClick?: (e: SyntheticEvent) => void;
 }> = ({ children, Component, userName, isButton, onClick, title }) => {
+  const encodedUserName = encodeURIComponent(userName);
+
   const { formatUrl, search } = useFormatUrl(SecurityPageName.users);
   const { navigateToApp } = useKibana().services.application;
   const goToUsersDetails = useCallback(
     (ev) => {
       ev.preventDefault();
       navigateToApp(APP_UI_ID, {
         deepLinkId: SecurityPageName.users,
-        path: getUsersDetailsUrl(encodeURIComponent(userName), search),
+        path: getUsersDetailsUrl(encodeURIComponent(encodedUserName), search),
       });
     },
-    [userName, navigateToApp, search]
+    [encodedUserName, navigateToApp, search]
   );
 
   return isButton ? (
     <GenericLinkButton
       Component={Component}
       dataTestSubj="data-grid-user-details"
-      href={formatUrl(getUsersDetailsUrl(encodeURIComponent(userName)))}
+      href={formatUrl(getUsersDetailsUrl(encodeURIComponent(encodedUserName)))}
```

Comment:
`encodedUserName === encodeURIComponent(userName)` so this line would be `encodeURIComponent(encodeURIComponent(userName))`. is that what we want?

Reply:
No no, it is duplicated. I am fixing it